### PR TITLE
Refactor Model::action to Model::toQuery

### PR DESCRIPTION
**Pull request by mvorisek**
- doc might need an update (search `(?<!\w)action(?!\w)`)
- `function action` is not renamed in persistences (called only in one place - Model::toQuery)
- Array persistence does not use query :) - how to solve it?

### CODE REVIEW

File: src/Model.php [hunk 1]
```diff
@@ -1881,7 +1881,7 @@ public function atomic($f, Persistence $persistence = null)
      *
      * @return Query
      */
-    public function action($mode, $args = [])
+    public function toQuery($mode, $args = [])
```
Comment (georgehristov) [inline src/Model.php:1883]: We should keep `action` forwarding to `toQuery` with a deprecation notice

**Comment by georgehristov** [comment]
Please rebase it on [fix/has-one-ref](https://github.com/atk4/data/tree/fix/has-one-ref) and include deprecation notice on the `Model::action` method

**Comment by georgehristov** [comment]
> Array persistence does not use query :) - how to solve it?

Implement `Persistence\QueryInterface` and require `Persistence::action` to return this interface.

It will have the methods in `Query` and we apply this to `Action/Iterator` (it needs to be moved out of there to `Persistence/Array_/Query`)
And we will have 
- `Persistence\Array_\Query implements Persistence\QueryInterface`.
- `Persistence\Sql\Query extends dsql\Query implements Persistence\QueryInterface`. Will be empty class :)

@mvorisek will you take it up?

File: src/Model.php (hunk 1 above)
Comment (DarkSide666) [reply src/Model.php:1883]: Yes this is a must because `action()` method is used in projects quite widely.
Also this PR probably need BC-break label.

Comment (mvorisek) [reply src/Model.php:1883]: labeled

**Comment by mvorisek** [comment]
closed in favor of https://github.com/atk4/data/pull/690

**Pull request closed without merging**